models/cam-fv/model_mod.f90

```diff
@@ -2380,6 +2383,30 @@ subroutine get_close_state(gc, base_loc, base_type, locs, loc_qtys, loc_indx, &
 
 end subroutine get_close_state
 
+!-----------------------------------------------------------------------
+
+subroutine verify_state_var_list
+
+! No return is needed.  A diagnostic statement is printed before a potential failure
+! of get_varid_from_varname.  Success means carry on.
+
+! There could be more moisture variables than the traditional 3.
+character(len=32), dimension(10) :: var_names
+integer:: i, varid
+
+data var_names /'PS', 9*''/
```

Comment:
I don't think you need a data statement & and i variable here
You only have 'PS' so you could call 

`varid = get_varid_from_varname(domain_id, 'PS')`